symengine/fields.cpp

```diff
@@ -462,4 +462,102 @@ GaloisFieldDict::gf_ddf_zassenhaus() const
     }
     return factors;
 }
+
+GaloisFieldDict
+GaloisFieldDict::_gf_pow_pnm1d2(const GaloisFieldDict &f,
+                                const integer_class &n,
+                                const std::vector<GaloisFieldDict> &b) const
+{
+    GaloisFieldDict f_in(f);
+    f_in %= *this;
+    GaloisFieldDict h, r;
+    h = r = f_in;
+    for (unsigned i = 1; i < n; ++i) {
+        h = h.gf_frobenius_map(*this, b);
+        r *= h;
+        r %= *this;
+    }
+    auto res = gf_pow_mod(r, (modulo_ - 1_z) / 2_z);
+    return res;
+}
+
+GaloisFieldDict GaloisFieldDict::gf_random(integer_class n) const
+{
+    int n_val = mp_get_si(n);
+    int mod_val = mp_get_si(modulo_);
+    std::vector<integer_class> v(n_val + 1);
+    for (int i = 0; i < n; ++i) {
+        v[i] = integer_class(std::rand() % mod_val);
```

Comment:
There are random number generators in GMP. We should use one of those if we are dealing with big numbers.